src/main/java/com/iota/iri/service/tipselection/impl/WalkValidatorImpl.java

```diff
@@ -75,19 +82,30 @@ public boolean isValid(Hash transactionHash) throws Exception {
         return true;
     }
 
-    private boolean belowMaxDepth(Hash tip, int depth) throws Exception {
+    private boolean belowMaxDepth(Hash tip, int allowedSnapshotIndex) throws Exception {
         //if tip is confirmed stop
-        if (TransactionViewModel.fromHash(tangle, tip).snapshotIndex() >= depth) {
+        if (TransactionViewModel.fromHash(tangle, tip).snapshotIndex() >= allowedSnapshotIndex) {
             return false;
         }
         //if tip unconfirmed, check if any referenced tx is confirmed below maxDepth
         Queue<Hash> nonAnalyzedTransactions = new LinkedList<>(Collections.singleton(tip));
         Set<Hash> analyzedTransactions = new HashSet<>();
         Hash hash;
         while ((hash = nonAnalyzedTransactions.poll()) != null) {
+            if (FAILED_BELOW_MAX_DEPTH_CACHE.contains(hash)) {
+                updateCache(analyzedTransactions);
+                return true;
+            }
+            if (analyzedTransactions.size() == MAX_ANALYZED_TXS) {
+                updateCache(analyzedTransactions);
+                return true;
+            }
```

Comment:
If you wanted to make it a bit more concise, I think you could combine these into one `if` statement:
```java
if (FAILED_BELOW_MAX_DEPTH_CACHE.contains(hash) || analyzedTransactions.size() == MAX_ANALYZED_TXS) {
  updateCache(analyzedTransactions);
  return true;
}
```
But I guess that's a matter of preference

Reply:
I agree with you actually